src/main/java/seedu/address/logic/graph/BfsHopCroftKarp.java

```diff
@@ -0,0 +1,156 @@
+     * path(s) is found, otherwise an empty list is returned. The given predecessors array are used to record the
+     * structure of the layered graph.
+     *
+     * @param intervieweePredecessor a list used to keep track of the predecessor of an interviewee, which is a
+     *                               slot matched to it.
+     * @param interviewSlotPredecessors a list used to keep track of predecessor(s) of an interview slot, which is
+     *                                  interviewee(s) that can match the interview slot.
+     */
+    public List<InterviewSlotVertex> execute(List<InterviewSlotVertex> intervieweePredecessor,
+                                             List<List<IntervieweeVertex>> interviewSlotPredecessors) {
+        Queue<Vertex> currentLayer = initialiseBfs();
+        Queue<Vertex> nextLayer = new LinkedList<>();
+        List<InterviewSlotVertex> unmatchedSlotVertices = new LinkedList<>();
+
+        // Perform BFS and build the layered graph
+        do {
+            nextLayer.clear();
+            // Constructs the next layer
+            while (!currentLayer.isEmpty()) {
+                Vertex u = currentLayer.poll();
+
+                if (u instanceof IntervieweeVertex) {
+                    nextLayer.addAll(findUnmatchedVertices(u, nextLayer, interviewSlotPredecessors));
+                } else if (u instanceof InterviewSlotVertex && u.isMatched()) {
+                    Vertex v = findMatchedVertex(u, nextLayer, intervieweePredecessor);
+                    if (v != null) {
+                        nextLayer.add(v);
+                    }
+                }
+            }
+
+            if (containsUnmatchedSlotVertices(nextLayer)) {
+                unmatchedSlotVertices = getUnmatchedSlotVertices(nextLayer);
+                break;
+            } else {
+                currentLayer.addAll(nextLayer);
+            }
+        } while (!nextLayer.isEmpty());
+
+        return unmatchedSlotVertices;
+    }
+
+    /**
+     * Returns a queue of unmatched vertices in the first layer of the graph
+     * (which is basically unmatched interviewee vertices).
+     */
+    private Queue<Vertex> initialiseBfs() {
+        int numInterviewees = graph.getNumInterviewees();
+        Queue<Vertex> firstLayer = new LinkedList<>();
+
+        // Initialise the first layer
+        for (int i = 0; i < numInterviewees; i++) {
+            IntervieweeVertex intervieweeVertex = graph.getIntervieweePair(i).getHead();
+
+            if (!intervieweeVertex.isMatched()) {
+                firstLayer.offer(intervieweeVertex);
+            }
+        }
+
+        return firstLayer;
+    }
+
+    /**
+     * Returns a list of unmatched interview slot vertices in the next layer.
+     */
+    private List<Vertex> findUnmatchedVertices(Vertex u, Queue<Vertex> nextLayer,
+                                               List<List<IntervieweeVertex>> interviewSlotPredecessors) {
```

Comment:
I think the indentation here is incorrect, should be 8 spaces more than the previous line